Nautilus/Patchers/VehicleUpgradesPatcher.cs

```diff
@@ -0,0 +1,522 @@
+    {
+        if (!VehicleUpgradeModules.TryGetValue(techType, out var prefab))
+            return;
+
+        if (!prefab.TryGetGadget(out UpgradeModuleGadget moduleGadget))
+            return;
+
+        if (moduleGadget.delegateOnRemoved != null && !added)
+            moduleGadget.delegateOnRemoved.Invoke(__instance, slotID);
+
+        if (moduleGadget.delegateOnAdded != null && added)
+            moduleGadget.delegateOnAdded.Invoke(__instance, slotID);
+    }
+
+    [HarmonyPrefix]
+    [HarmonyPatch(typeof(Vehicle), nameof(Vehicle.OnUpgradeModuleChange))]
+    private static void OnModuleChangeCrushDepth(Vehicle __instance, int slotID, TechType techType, bool added)
+    {
+        Dictionary<TechType, float> CrushDepthUpgrades = new();
+        VehicleUpgradeModules.DoIf(
+            (KeyValuePair<TechType, ICustomPrefab> mapElem) => mapElem.Value.TryGetGadget(out UpgradeModuleGadget moduleGadget) && moduleGadget.CrushDepth > 0f,
+            (KeyValuePair<TechType, ICustomPrefab> mapElem) => CrushDepthUpgrades.Add(mapElem.Key, mapElem.Value.GetGadget<UpgradeModuleGadget>().CrushDepth)
+        );
+
+        var newCrushDepth = 0f;
+        var absolute = false;
+        for (var i = 0; i < __instance.slotIDs.Length; i++)
+        {
+            string slot = __instance.slotIDs[i];
+            TechType techTypeInSlot = __instance.modules.GetTechTypeInSlot(slot);
+            if (CrushDepthUpgrades.ContainsKey(techTypeInSlot))
+            {
+                float crushDepthToCheck = CrushDepthUpgrades[techTypeInSlot];
+                if (crushDepthToCheck > newCrushDepth)
+                {
+                    newCrushDepth = crushDepthToCheck;
+                    if (VehicleUpgradeModules[techTypeInSlot].TryGetGadget(out UpgradeModuleGadget moduleGadget))
+                        absolute = moduleGadget.AbsoluteDepth;
+                }
+            }
+        }
+        if (absolute)
+            __instance.crushDamage.SetExtraCrushDepth(newCrushDepth - __instance.crushDamage.kBaseCrushDepth);
+        else
+            __instance.crushDamage.SetExtraCrushDepth(newCrushDepth);
+    }
+
+    //
+    // VEHICLE
+    // LAZY INITIALIZE (OnToggle)
+    //
+    [HarmonyPostfix]
+    [HarmonyPatch(typeof(Vehicle), nameof(Vehicle.LazyInitialize))]
+    private static void LazyInitialize(Vehicle __instance)
+    {
+        __instance.onToggle += (int slotID, bool state) =>
+        {
+            ICustomPrefab prefab;
+            var techType = __instance.modules.GetTechTypeInSlot(__instance.slotIDs[slotID]);
+            UpgradeModuleGadget moduleGadget;
+            double energyCost;
+            switch (__instance)
+            {
+                case Exosuit:
+                    break;
+                case SeaMoth:
+                    if (techType == TechType.None)
+                        break;
+
+                    if (!SeamothUpgradeModules.TryGetValue(techType, out prefab))
```

Comment:
Causes a compiler error on the Below Zero build target.

Reply:
Ah yeah I must put that in regions my bad